### DISCUSSION TITLE
It is possible to have two injected wallets like MetaMask and TrustWallet extension?

### DISCUSSION THREAD

**Discussion by mshadev** [wevm/wagmi#2777] [Q&A]
Hi! 

Let me explain my situation: 

I have installed two extensions: MetaMask and TrustWallet.
On the webpage, I have two buttons. One button should trigger the MetaMask extension, and the other TrustWallet. 
But now, triggered only last installed extension. In my case it is TrustWallet.

Is it possible to make only the correct extension run depending on which button is clicked?

Thanks

**Comment by glitch-txs** [comment]
yes, you need to use injected Connector and add a getProvider function specially for Trust Wallet. They inject in `window.trustwallet`

**Reply by trishagonzales** [reply]
how about in trustwallet mobile? how do i fetch the provider?

**Reply by glitch-txs** [reply]
for mobile use WalletConnect
